bundle/Controller/UserController.php

```diff
@@ -0,0 +1,168 @@
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+namespace EzSystems\RepositoryFormsBundle\Controller;
+
+use eZ\Bundle\EzPublishCoreBundle\Controller;
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\Exceptions\UnauthorizedException;
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\API\Repository\UserService;
+use eZ\Publish\Core\Base\Exceptions\BadStateException;
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentType;
+use EzSystems\RepositoryForms\Data\Mapper\UserCreateMapper;
+use EzSystems\RepositoryForms\Data\Mapper\UserUpdateMapper;
+use EzSystems\RepositoryForms\Form\ActionDispatcher\ActionDispatcherInterface;
+use EzSystems\RepositoryForms\Form\Type\User\UserCreateType;
+use EzSystems\RepositoryForms\Form\Type\User\UserUpdateType;
+use EzSystems\RepositoryForms\User\View\UserCreateView;
+use EzSystems\RepositoryForms\User\View\UserUpdateView;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\OptionsResolver\Exception\AccessException;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Exception\MissingOptionsException;
+use Symfony\Component\OptionsResolver\Exception\NoSuchOptionException;
+use Symfony\Component\OptionsResolver\Exception\OptionDefinitionException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+
+class UserController extends Controller
+{
+    /** @var ContentTypeService */
+    private $contentTypeService;
+
+    /** @var UserService */
+    private $userService;
+
+    /** @var LocationService */
+    private $locationService;
+
+    /** @var ActionDispatcherInterface */
+    private $userActionDispatcher;
+
+    public function __construct(
+        ContentTypeService $contentTypeService,
+        UserService $userService,
+        LocationService $locationService,
+        ActionDispatcherInterface $userActionDispatcher
+    ) {
+        $this->contentTypeService = $contentTypeService;
+        $this->userService = $userService;
+        $this->locationService = $locationService;
+        $this->userActionDispatcher = $userActionDispatcher;
+    }
+
+    /**
+     * Displays and processes a user creation form.
+     *
+     * @param string $contentTypeIdentifier ContentType id to create
+     * @param string $language Language code to create the content in (eng-GB, ger-DE, ...))
+     * @param int $parentLocationId Location the content should be a child of
+     * @param Request $request
+     *
+     * @return UserCreateView|Response
+     *
+     * @throws InvalidArgumentType
```

Comment:
Ain't that a lot of exceptions for such a little method ? :)
I'm wondering if we shouldn't catch and group most of these into exceptions that make sense from the standpoint of a controller.